src/dmd/dtemplate.d

```diff
@@ -1609,7 +1609,7 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
 
                     // https://issues.dlang.org/show_bug.cgi?id=12876
                     // Optimize argument to allow CT-known length matching
-                    farg = farg.optimize(WANTvalue, (fparam.storageClass & (STC.ref_ | STC.out_)) != 0);
+                    farg = farg.optimize(WANTvalue, fparam.isRefOrOut() != 0);
```

Comment:
remove the `!= 0`

Reply:
in this case (and probably the others too) `isRefOrOut` returns a `StorageClass`, i.e `ulong`, which does not implicitly converts to `bool`.

Reply:
`final bool isRefOrOut()` am I missing something?

Reply:
confirmed for all the other use of `!= 0`.

Reply:
>  am I missing something?

Yes the helper is implemented twice for `class Declaration` and `class Parameter`. IN one case it returns `bool`, the other `StorageClass` because at least one of the OrExpression that's refactored was used to get a `StorageClass` (for a class in hdrgen).

Reply:
can we try to be consistent? If the storage class is needed then I think it would be best to return it all the time. Or name the one that returns a bool differently.

Reply:
If you return a bool, use the `isWhatever` convention.  If you return a different type, consider the `asWhatever` convention.